src/sage/rings/polynomial/polynomial_ring_constructor.py

```diff
@@ -804,7 +810,11 @@ def _single_variate(base_ring, name, sparse=None, implementation=None, order=Non
             constructor = polynomial_ring.PolynomialRing_integral_domain
         else:
             constructor = polynomial_ring.PolynomialRing_commutative
-        implementation_names = constructor._implementation_names(implementation, base_ring, sparse)
+
+        if isinstance(base_ring, TropicalSemiring):
+            implementation_names = [None]
+        else:
+            implementation_names = constructor._implementation_names(implementation, base_ring, sparse)
```

Comment:
Instead of changing this, you should change the `TropicalPolynomialSemiring` to match the other implementation with the `@classmethod`.